### PR TITLE
Reference page changes

### PR DESCRIPTION
**Pull request by volodymyrmakarchuk-okta**
## Description:
- **What's changed?** Refactor reference page. Add sidebar and page nav tree 
- **Is this PR related to a Monolith release?** 

### Resolves:

* [OKTA-351801](https://oktainc.atlassian.net/browse/OKTA-351801)

### CODE REVIEW

**Review by alexdahl-okta — APPROVED** [review]
UX looks good, thanks! Docs team and engineering should review as well.

File: packages/@okta/vuepress-site/docs/reference/index.md [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
+
+
+Okta APIs generally fall two categories: signing in your users, and managing
+resources in your Okta org.
+
 ## Sign in Your Users
 API endpoints to authenticate your users, challenge for factors, recover passwords, and more. [Learn about which APIs to use.](/docs/concepts/oauth-openid/#authentication-api-vs-oauth-20-vs-openid-connect)
+API endpoints to authenticate your users, challenge for factors, recover passwords, and more.
+  - The [Authentication API](/docs/reference/api/authn) controls user accesss to Okta.
+  - The [OpenID Connect & OAuth 2.0 API](/docs/reference/api/oidc/) controls users access to your applications.
```
Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/reference/index.md:17]: ... controls **user** access to your applications.

File: packages/@okta/vuepress-site/docs/reference/index.md [hunk 2]
```diff
@@ -1,31 +1,25 @@
...
-::: slot left
+# Reference overview
+
+Details on parameters, requests, and responses for Okta's API endpoints.
+
+
+Okta APIs generally fall two categories: signing in your users, and managing
+resources in your Okta org.
+
 ## Sign in Your Users
```
Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/reference/index.md:13]: Sign in your users

(Sentence style capitalization in headers)

File: packages/@okta/vuepress-site/docs/reference/index.md [hunk 3]
```diff
@@ -1,31 +1,25 @@
...
+  - The [OpenID Connect & OAuth 2.0 API](/docs/reference/api/oidc/) controls users access to your applications.
 
-<CategoryLinks category="authentication" class="list--with-descriptions" />
-:::
+[Learn about which APIs to use.](/docs/concepts/oauth-openid/#authentication-api-vs-oauth-20-vs-openid-connect)
 
-::: slot right
-## Manage Okta Resources
-REST endpoints to configure resources such as users, apps, sessions, and factors whenever you need.
+## Manage Okta Objects
```
Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/reference/index.md:21]: Manage Okta objects

File: packages/@okta/vuepress-site/docs/reference/index.md [hunk 4]
```diff
@@ -1,31 +1,25 @@
...
+
+Details on parameters, requests, and responses for Okta's API endpoints.
+
+
+Okta APIs generally fall two categories: signing in your users, and managing
+resources in your Okta org.
+
 ## Sign in Your Users
 API endpoints to authenticate your users, challenge for factors, recover passwords, and more. [Learn about which APIs to use.](/docs/concepts/oauth-openid/#authentication-api-vs-oauth-20-vs-openid-connect)
+API endpoints to authenticate your users, challenge for factors, recover passwords, and more.
```
Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/reference/index.md:15]: API endpoints ~to~ authenticate your users, challenge for factors, recover passwords, and more.

(Clarifies what the API endpoints can do)

File: packages/@okta/vuepress-site/docs/reference/index.md [hunk 5]
```diff
@@ -1,31 +1,25 @@
...
 
-<CategoryLinks category="authentication" class="list--with-descriptions" />
-:::
+[Learn about which APIs to use.](/docs/concepts/oauth-openid/#authentication-api-vs-oauth-20-vs-openid-connect)
 
-::: slot right
-## Manage Okta Resources
-REST endpoints to configure resources such as users, apps, sessions, and factors whenever you need.
+## Manage Okta Objects
+REST endpoints to configure resources such as users, apps, sessions, and factors
```
Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/reference/index.md:22]: REST endpoints ~to~ configure resources such as users, apps, sessions, and factors when needed.

File: packages/@okta/vuepress-site/docs/reference/index.md [hunk 6]
```diff
@@ -1,31 +1,25 @@
...
+whenever you need.
```
Comment (IanHakes-okta) [inline packages/@okta/vuepress-site/docs/reference/index.md:23]: ~whenever you need~ 
(removed in previous comment)

File: packages/@okta/vuepress-site/docs/reference/index.md (hunk 1 above)
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-site/docs/reference/index.md]: done

File: packages/@okta/vuepress-site/docs/reference/index.md (hunk 2 above)
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-site/docs/reference/index.md]: done

File: packages/@okta/vuepress-site/docs/reference/index.md (hunk 3 above)
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-site/docs/reference/index.md]: done

File: packages/@okta/vuepress-site/docs/reference/index.md (hunk 4 above)
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-site/docs/reference/index.md]: done

File: packages/@okta/vuepress-site/docs/reference/index.md (hunk 5 above)
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-site/docs/reference/index.md]: done

File: packages/@okta/vuepress-site/docs/reference/index.md (hunk 6 above)
Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-site/docs/reference/index.md]: done

**Review by IanHakes-okta — APPROVED** [review]
LGTM

**Review by shawyu-okta — APPROVED** [review]
Moved base branch to feature branch FYI

lgtm 🚀

**Review by francoislehoux-okta — APPROVED** [review]
LGTM

**Pull request merged**